crates/sui-core/src/transaction_driver/request_retrier.rs

```diff
@@ -42,19 +43,16 @@ impl<A: Clone> RequestRetrier<A> {
         tx_type: TxType,
         allowed_validators: Vec<AuthorityName>,
     ) -> Self {
-        let selected_validators = if !allowed_validators.is_empty() {
-            allowed_validators
-        } else {
-            client_monitor.select_shuffled_preferred_validators(
-                &auth_agg.committee,
-                auth_agg.committee.num_members() / TOP_K_VALIDATORS_DENOMINATOR,
-                tx_type,
-            )
-        };
-        let remaining_clients = selected_validators
+        let ranked_validators = client_monitor.select_shuffled_preferred_validators(
+            &auth_agg.committee,
+            tx_type,
+            SELECT_LATENCY_DELTA,
+        );
+        let ranked_clients = ranked_validators
             .into_iter()
+            .filter(|name| allowed_validators.contains(name))
```

Comment:
`allowed_validators` can be an empty vector (if there are no restrictions)

```suggestion
            .filter(|name| allowed_validators.is_empty() || allowed_validators.contains(name))
```